h/static/scripts/group-forms/hooks/test/use-update-moderation-status-test.js

```diff
@@ -0,0 +1,101 @@
+import { mount } from '@hypothesis/frontend-testing';
+
+import { Config } from '../../config';
+import {
+  useUpdateModerationStatus,
+  $imports,
+} from '../use-update-moderation-status';
+
+describe('useUpdateModerationStatus', () => {
+  let fakeConfig;
+  let fakeAnnotation;
+  let fakeCallAPI;
+  let lastUpdateModerationStatus;
+
+  beforeEach(() => {
+    fakeConfig = { api: {} };
+    fakeAnnotation = {
+      id: '1',
+      updated: '2025-06-24T08:43:50.914807+00:00',
+    };
+
+    fakeCallAPI = sinon.stub();
+
+    $imports.$mock({
+      '../utils/api': { callAPI: fakeCallAPI },
+    });
+
+    lastUpdateModerationStatus = undefined;
+  });
+
+  function TestComponent({ annotation }) {
+    lastUpdateModerationStatus = useUpdateModerationStatus(annotation);
+    return null;
+  }
+
+  function createComponent(annotation = fakeAnnotation) {
+    return mount(
+      <Config.Provider value={fakeConfig}>
+        <TestComponent annotation={annotation} />
+      </Config.Provider>,
+    );
+  }
+
+  it('does not call API if config is not available', () => {
+    createComponent();
+    lastUpdateModerationStatus('APPROVED');
+
+    assert.notCalled(fakeCallAPI);
+  });
+
+  it('does not call API if annotation does not have an ID', () => {
```

Comment:
If you change the `APIAnnotationData` type to reflect the fact that annotations will always have an ID, then this test won't be required. If in future we add the ability to create new, unsaved annotations in h, then at that point we may need to introduce a new type to represent unsaved annotations.